src/coreclr/jit/codegenarmarch.cpp

```diff
@@ -1879,6 +1879,547 @@ void CodeGen::genCodeForCpBlkHelper(GenTreeBlk* cpBlkNode)
     }
 }
 
+#ifdef TARGET_ARM64
+
+class CountingStream
+{
+public:
+    CountingStream()
+    {
+        instrCount = 0;
+    }
+
+    void LoadPairRegs(int offset, int regSizeBytes)
+    {
+        instrCount++;
+    }
+
+    void StorePairRegs(int offset, int regSizeBytes)
+    {
+        instrCount++;
+    }
+
+    void LoadReg(int offset, int regSizeBytes)
+    {
+        instrCount++;
+    }
+
+    void StoreReg(int offset, int regSizeBytes)
+    {
+        instrCount++;
+    }
+
+    int InstructionCount() const
+    {
+        return instrCount;
+    }
+
+private:
+    int instrCount;
+};
+
+class StoreBlockUnrollHelper
```

Comment:
It seems odd that this helper is placed between two "Stream" classes; maybe it should be hoisted up?

Also, it's used by the InitBlock code as well, so maybe the name should be more generally `BlockUnrollHelper`?